src/assets/wise5/components/table/table-authoring/table-authoring.component.ts

```diff
@@ -308,21 +313,62 @@ export class TableAuthoring extends ComponentAuthoring {
     if (cellSize == '') {
       cellSize = null;
     }
-    this.setColumnCellSizes(index, cellSize);
+    this.setColumnCellSize(index, cellSize);
   }
 
-  setColumnCellSizes(column: number, size: number): void {
+  setColumnCellSize(column: number, size: number): void {
     const tableData = this.authoringComponentContent.tableData;
-    for (let r = 0; r < tableData.length; r++) {
-      const row = tableData[r];
-      const cell = row[column];
+    const firstRow = tableData[0];
+    if (firstRow != null) {
+      const cell = firstRow[column];
       if (cell != null) {
         cell.size = size;
       }
     }
     this.componentChanged();
   }
 
+  private parseFrozenColumns(componentContent: any): any {
+    const frozenColumns = {};
+    const firstRow = componentContent.tableData[0];
+    if (firstRow != null) {
+      for (const key in firstRow) {
+        const cell = firstRow[key];
+        frozenColumns[key] = cell.frozen;
+      }
+    }
+    this.frozenColumnsLimitReached = this.isfrozenColumnsLimitReached();
+    return frozenColumns;
+  }
+
+  frozenColumnsChanged(index: number): void {
+    let frozen = this.frozenColumns[index];
+    const firstRow = this.authoringComponentContent.tableData[0];
+    if (firstRow != null) {
+      const cell = firstRow[index];
+      if (cell != null) {
+        cell.frozen = frozen;
+      }
+    }
+    this.componentChanged();
+    this.frozenColumnsLimitReached = this.isfrozenColumnsLimitReached();
+  }
+
+  private isfrozenColumnsLimitReached(): boolean {
+    let count = 0;
+    const firstRow = this.authoringComponentContent.tableData[0];
+    if (firstRow == null) {
+      return false;
+    }
+    const maxFrozen = firstRow.length - 1;
+    for (const key in firstRow) {
+      if (firstRow[key].frozen) {
+        count++;
+      }
+    }
+    return count === maxFrozen;
+  }
```

Comment:
Move variable declarations just before you need them
```suggestion
  private isfrozenColumnsLimitReached(): boolean {
    const firstRow = this.authoringComponentContent.tableData[0];
    if (firstRow == null) {
      return false;
    }
    let count = 0;
    for (const key in firstRow) {
      if (firstRow[key].frozen) {
        count++;
      }
    }
    const maxFrozen = firstRow.length - 1;
    return count === maxFrozen;
  }
```